### ISSUE TITLE
Would you be interested to contribute your module ?

### ISSUE THREAD

**Issue by max3903** [slmoebius/builder#2]
Hello,

Have you check module_prototyper ? It is being reviewed here:
https://github.com/OCA/server-tools/pull/107

and the roadmap is available here:
https://github.com/OCA/server-tools/issues/104

Would you be interested to converge and merge our modules into one ?

**Comment by slmoebius** [comment]
Yes sure!, I'll take a look at it and see common ground.

**Comment by slmoebius** [comment]
Thanks for your interest in our module, we are excited about it. We looked out your module and seemed to us that we have different approaches. Our module focus on the creation of new modules completely detached from the current installation, and the models, views, groups, etc are applied to the system only when you download and install the module, while yours apply changes directly in the database and then export them as a new module. We understand your point of view and we think we can include it in our module by creating importers for each element you manage. 
We are very interested in join efforts with you to create a more complete tool for code generation of Odoo modules but our module is more complete right now and some of the roadmap you’re planning to do is already in the module builder. Could it be possible to take our module as the base for the development? Let us know what you think.

**Comment by max3903** [comment]
Yes, we do apply the customizations directly in the database so the user can have a better idea of how it will look like. We then export them into a module to ease the job of a developer who can focus on implementing the logic.

It also fit the scenario where a lot of customization has been done in the database directly and we want to export them into a module for easier migration.

I have no problem taking your module as the base for future developments. On the other hand, would you mind signing the OCA CLA and contributing it to https://github.com/OCA/server-tools ?

**Comment by max3903** [comment]
What's your opinion regarding the CLA ? Any chance to sign it ?

**Comment by yunier-rojas** [comment]
Hi there.
After thinking you proposal, we believe that contributing to OCA is the best for builder and for Odoo community. Let us know what's next. Cheers.

**Comment by bigdawgbz** [comment]
I was just doing the todo_app module from the Odoo Essentials Book..

**Comment by max3903** [comment]
@yunier-rojas, @slmoebius I will be presenting the prototyper next week in Belgium:
https://www.odoo.com/fr_FR/event/odoo-experience-2015-edition-2015-06-03-2015-06-05-304/track/prototype-your-module-with-odoo-250
so we spent some time to make sure I have a working demo/presentation.

But if you are in Belgium, maybe we can meet and work on converging. I will arrive on sunday and take part in the OCA code sprint: https://odoo-community.org/blog/news-1/post/oca-sprint-june-1-2-2015-18

I would like to understand the reason why you recreated all the tables for modules, models, fields, etc. Maybe it brings something I don't see. My email : maxime DOT chambreuil AT savoirfairelinux DOT com

Regards,

**Comment by onemoverx** [comment]
@bigdawgbz out of place but glat to hear you are trying builder for that module example, please let us know how it went.

**Comment by onemoverx** [comment]
@max3903 we contacted you by email but I wasn´t sure if you get it. 
the answer about why we did builder this way is this:
¨We conceive builder with almost the same models as odoo to keep the same structure and be able to extend it as the module grows, also make it independent of the odoo installation, that allow us to export/import the modules being created in different databases. All the changes are made in one place which give you a good overview of what you're doing. We are aware of some of the disadvantages too, like the unability to view the changes as it happens.¨

Right now there are some bugs on the generation that will be resolved with the next update alone with some other cool features and improvements.

**Comment by max3903** [comment]
@onemoverx I was out of the office last week at the Odoo Experience 2015. I have received your email. The subject is a hot topic and interests a lot of people. Would you be interested to work on the prototyper if we take care of porting all the current feature of the builder ? Being able to make changes and test them before exporting is a great advantage. Creating a new database to test the module content (views, models) is time-consuming.

**Comment by bigdawgbz** [comment]
I would love too. Not all that knowledgeable in Odoo, I am just starting to
play with it. How to be able to do my own modules soon.

**Comment by onemoverx** [comment]
I disagree on the time consuming point because its easy to go to the database manager and clone a database and test the generated module there, although I do think that it's quite a job download it and place it in the module folder and upgrade it. Thinking in that we made another module (not sure if we pushed the latest version into the github repository) to install the module in the same way that modules are uploaded (using the module_import module if I recall right). About working on the prototyper I do agree but we have so little time right now and we didn't even get to do the planned developments on the module builder that we wanted to, but count me in and I'll do what I can.
